plutus-scb/src/Cardano/Metadata/Client.hs

```diff
@@ -12,44 +12,47 @@
 module Cardano.Metadata.Client
     ( handleMetadataClient
     ) where
+
 import           Cardano.Metadata.API      (API)
-import           Cardano.Metadata.Types    (JSONEncoding (ExternalEncoding),
-                                            MetadataEffect (GetProperties, GetProperty),
+import           Cardano.Metadata.Types    (JSONEncoding (AesonEncoding),
+                                            MetadataEffect (BatchQuery, GetProperties, GetProperty),
                                             MetadataError (MetadataClientError))
 import           Control.Monad.Freer       (Eff, LastMember, Member, type (~>), interpret, sendM)
 import           Control.Monad.Freer.Error (Error, throwError)
 import           Control.Monad.IO.Class    (MonadIO, liftIO)
-import           Data.Coerce               (coerce)
+import           Data.Coerce               (Coercible, coerce)
 import           Data.Proxy                (Proxy (Proxy))
 import           Network.HTTP.Types        (status404)
-import           Servant.Client            (ClientEnv, ClientError (FailureResponse), ClientM, client,
+import           Servant.Client            (Client, ClientEnv, ClientError (FailureResponse), ClientM, client,
                                             responseStatusCode, runClientM)
 import           Servant.Extra             (left, right)
 
+-- Maintainer's Note: The api can be flipped between 'AesonEncoding'
+-- and 'ExternalEncoding' depending on which metadata server we're
+-- using. See the notes on this, and 'coerce', in
+-- 'Cardano.Metadata.Types'.
+api :: Client ClientM (API 'AesonEncoding)
+api = client (Proxy @(API _))
+
 handleMetadataClient ::
        forall m effs.
        (LastMember m effs, MonadIO m, Member (Error MetadataError) effs)
     => ClientEnv
     -> Eff (MetadataEffect ': effs) ~> Eff effs
 handleMetadataClient clientEnv =
-        -- The api can be flipped between 'AesonEncoding' and
-        -- 'ExternalEncoding' depending on which metadata server
-        -- we're using. See the notes on this, and 'coerce', in
-        -- 'Cardano.Metadata.Types'.
-    let api = client (Proxy @(API 'ExternalEncoding))
-
-        (getProperties, getProperty) = (left . api, right . api)
-
-        runClient :: ClientM (Maybe a) -> Eff effs (Maybe a)
-        runClient action = do
+    let getProperties subject = left (left api subject)
```

Comment:
Because this is the neatest way I've found of unpacking a servant `client` when you have `Capture` arguments feeding into some of the functions (but not all of them). Especially when two endpoints share a subset of those arguments.


In other words, you hope you can write:

```haskell
((getProperties :<|> getProperty) :<|> batchQuery) = client
```

...but what you actually need to say is closer to:

```haskell
((\subject -> getProperties subject :<|> getProperty subject) :<|> batchQuery) = client
```

But you can't pattern-match a lambda on the left-hand side, so that would be a syntax error. 

If there's a better way to do it, I'd love to learn it. :-)

Reply:
I'm going to look into this separately. If there is a better approach, it'll be in a separate PR.